tests/test_cli_event_source_guards.py

```
@@ -0,0 +1,111 @@
+def validate_socket_error_context_precedence(source: str) -> None:
+    start = source.index("    func captureError(")
+    end = source.index("        let subcommand = self.subcommand", start)
+    body = source[start:end]
+    generic_context_merge = body.index("        for (key, value) in data {")
+    socket_error_context_merge = body.index(
+        "        if let connectError = error as? CLISocketConnectError {"
+    )
+    if socket_error_context_merge < generic_context_merge:
+        raise AssertionError(
+            "generic operation data can overwrite typed socket error diagnostics"
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)